guidance/guidance_los/guidance_los/los_guidance_algorithm.py

```diff
@@ -1,28 +1,37 @@
-#!/usr/bin/env python3
-
 import numpy as np
 
 
 class ThirdOrderLOSGuidance:
-    """Line-of-Sight (LOS) guidance algorithm."""
+    """This class implements the Line-of-Sight (LOS) guidance algorithm.
+
+    The LOS provide a control outputs for surge, pitch, and yaw for navigating towards a target in 3D space.
+    """
 
     def __init__(self, config_dict: dict):
-        # Initialize parameters
-        self.h_delta_min = config_dict['h_delta_min']
-        self.h_delta_max = config_dict['h_delta_max']
-        self.h_delta_factor = config_dict['h_delta_factor']
-        self.nominal_speed = config_dict['nominal_speed']
-        self.min_speed = config_dict['min_speed']
-        self.max_pitch_angle = config_dict['max_pitch_angle']
-        self.depth_gain = config_dict['depth_gain']
-
-        # Filter parameters
+        self.config_dict = config_dict
+        self.init_los_parameters()
+        self.init_filter_parameters()
+        self.setup_filter_matrices()
+
+    def init_los_parameters(self):
+        """Initialize Line-of-Sight guidance parameters."""
+        self.h_delta_min = self.config_dict['h_delta_min']
+        self.h_delta_max = self.config_dict['h_delta_max']
+        self.h_delta_factor = self.config_dict['h_delta_factor']
+        self.nominal_speed = self.config_dict['nominal_speed']
+        self.min_speed = self.config_dict['min_speed']
+        self.max_pitch_angle = self.config_dict['max_pitch_angle']
+        self.depth_gain = self.config_dict['depth_gain']
```

Comment:
Maybe a better option would be to input a dataclass with the LOS parameters instead of a dictionary, since it is more rigid and is able to use default values. So something like

```python
@dataclass(slots=True)
class LOSParams:
    h_delta_min ...
    h_delta_max ...
    h_delta_factor ...
    nominal_speed ...
    min_speed ...
    max_pitch_angle ...
    depth_gain ...
```
Also, might be a good idea to use a more descriptive name for `h`, since it is not immediately clear what it represents.